test/integration/scheduler_perf/util.go

```diff
@@ -203,21 +206,33 @@ func (*metricsCollector) run(ctx context.Context) {
 func (pc *metricsCollector) collect() []DataItem {
 	var dataItems []DataItem
 	for _, metric := range pc.Metrics {
-		dataItem := collectHistogramVec(metric, pc.labels)
-		if dataItem != nil {
-			dataItems = append(dataItems, *dataItem)
+		metricFamily, err := testutil.GetMetricFamilyFromGatherer(legacyregistry.DefaultGatherer, metric)
+		if err != nil {
+			klog.Error(err)
+			return dataItems
+		}
+		if metric == extentionPointsMetricName {
```

Comment:
I'm not sure I like this "hardcoded" comparison. Our goal is to make the metric format extensible, which shouldn't be restricted to the extension point.

You can define the metric as a string or map.

Reply:
removed "hardcoded" comparison here, and fetch the value from the map instead.